Add release note templates for future releases

sf1919 opened the pull request:
**Description of work.**
These are the new templates that will be used to generate the release notes for the next release. Previously the templates for the separate release notes were hard coded into release.py. However the changes to the release note process mean that this is no longer a suitable method. By having separate templates this will make it easier if changes need to be made in future.

The conf.py file has been updated to exclude the templates from the toc_tree as otherwise it is likely to cause build failures (see #33599)

**To test:**

- check that the headings for all except Muons in the templates match those listed in the Release Note Guide (https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)
- check that the headings for Muons match those listed in the revised file structure for Muons (see #33672 for details)
- check that each heading has an amalgamate sphinx directive with it.
- There should be no developer warnings or release note version information in the templates

Part of the release note project (#33593)

*This does not require release notes* because **this is not a user facing part of release notes**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Created initial setup files and script

- Added templates for main release notes
- Added script to create main files for a release, create relevant directories and subfiles

sf1919 committed:
Added script to collate release notes during development

sf1919 committed:
Added functionality for start of release

- Added script for moving release notes into main files
- Script includes code to moved used files into separate folders to denote they've been amalgamated
- Updated template for Direct Geometry as links were incorrect

sf1919 committed:
Added templates for release notes

sf1919 committed:
Updated conf file to omit templates from toc

sf1919 committed:
Remove obsolete scripts

SilkeSchomann commented on docs/source/release/templates/framework.rst:
```diff
@@ -0,0 +1,61 @@
+.. amalgamate:: Framework/Python/Bugfixes
+
+
+Installation
```
Installation is not mentioned in the Release Note Guide

sf1919 replied:
@martyngigg I'm not sure this is a heading we really use. Should we remove it from the template or leave it?

sf1919 replied:
Ok so I've thought about this and I'm just going to remove Installation from the templates. We can always add it when we want to use it but I just don't feel its needed to be there by default.

martyngigg replied:
Sorry for the delay in reply!
We generally don't affect the installation much so as a general rule it probably doesn't deserve being there as standard so I agree removing from the template makes sense.

SilkeSchomann commented on docs/source/release/templates/framework.rst:
```diff
@@ -0,0 +1,61 @@
+Installation
+------------
+
+MantidWorkbench
```
MantidWorkbench is not mentioned in the Release Note Guide

sf1919 replied:
The Framework release notes have had a link over to the Workbench release notes for some time. As the heading is for a link rather than any release notes it shouldn't be in the Release Note Guide. The guide should only include the relevant file structure.

SilkeSchomann commented on docs/source/release/templates/diffraction.rst:
```diff
@@ -0,0 +1,41 @@
+.. amalgamate:: Diffraction/Engineering/Bugfixes
+
+
+Single Crystal Diffraction
```
In the Release Note Guide this section is a second Powder Diffraction. I guess this template is correct and the Release Note Guide wrong though

sf1919 replied:
The release notes are definitely incorrect! So glad you spotted this. I'll get it updated as part of this PR.

sf1919 committed:
Changes from review

- Removed Installation heading from Framework
- Updated file list in Release Note Guide

SilkeSchomann reviewed (approved):
The templates look good and match the headings in the Release Note Guide.
